### PR TITLE
refactor(upload-operator): add generic to upload operator, return htt…

### PR DESCRIPTION
**Pull request by OleksiiSmerdov**
…p response body

### CODE REVIEW

File: projects/ngx-operators/src/lib/upload.ts [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
         state: "IN_PROGRESS",
+        body: null,
       };
     }
-    if (isHttpResponse(event)) {
+    if (isHttpResponse<T>(event)) {
       return {
         progress: 100,
         state: "DONE",
+        body: (event as HttpResponse<T>).body,
```
Comment (nilsmehlhorn) [inline projects/ngx-operators/src/lib/upload.ts:30]: Wondering if this type assertion is necessary as `isHttpResponse`  looks like it narrows to `HttpResponse<T>` already. :thinking:

File: projects/ngx-operators/src/lib/upload.ts [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
   state: "PENDING" | "IN_PROGRESS" | "DONE";
+  body: T | null;
 }
 
-export function upload(): (
-  source: Observable<HttpEvent<unknown>>
-) => Observable<Upload> {
-  const initialState: Upload = { state: "PENDING", progress: 0 };
-  const reduceState = (state: Upload, event: HttpEvent<unknown>): Upload => {
+export function upload<T>(): (
```
Comment (nilsmehlhorn) [inline projects/ngx-operators/src/lib/upload.ts:12]: How about we default the generic parameter to `unknown`? That way this wouldn't be a breaking change for users of the library

```suggestion
export function upload<T = unknown>(): (
```

**Review by nilsmehlhorn — CHANGES_REQUESTED** [review]
Thanks for the contribution, I've just got two minor comments.

File: projects/ngx-operators/src/lib/upload.ts (hunk 1 above)
Comment (OleksiiSmerdov) [reply projects/ngx-operators/src/lib/upload.ts]: Nope, it's not necessary. Fixed

File: projects/ngx-operators/src/lib/upload.ts (hunk 2 above)
Comment (OleksiiSmerdov) [reply projects/ngx-operators/src/lib/upload.ts]: This makes sense. Fixed

**Review by nilsmehlhorn — APPROVED** [review]

**Pull request merged**